fix: kubernetes installation, high availability and publish to internet docs based on usability testing findings (#1526)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx

4. A running Kubernetes cluster. If not present, follow these steps to set up the cluster:

    a. Verify if you have access to AWS EKS with:
      
    ```bash
      aws sts get-caller-identity 
    ```
    
    The above command provides information related to your account and ARN, indicating that you can connect and access your Amazon account using a terminal.

    b. Create KubeConfig with: 

    ```bash
    aws eks update-kubeconfig --region ap-south-1 --name CLUSTER_NAME  --profile PROFILE_NAME
    ```
    In the above command, add the profile name that has access to the EKS cluster to the `--profile` parameter.

    c. Test your Kubernetes configuration with:

    ```bash
    kubectl cluster-info
    ```
    The above command provides a summary of the current cluster configuration, including the Kubernetes master and other cluster information. 

    d. A default storage class running. If not created yet, create the storage class with:

    ```bash
    kubectl apply -f - <<EOF
      apiVersion: storage.k8s.io/v1
      kind: StorageClass
      apiVersion: storage.k8s.io/v1
      metadata:
        name: gp2

>       provisioner: kubernetes.io/aws-ebs

Reviewer: ```suggestion
      provisioner: kubernetes.io/aws-ebs
      EOF
```

Author: @pratapaprasanna - this looks the same to me.

<VideoEmbed host="youtube" videoId="wZzYL1uZwds" title="How To Deploy Appsmith On Kubernetes" caption="How To Deploy Appsmith On Kubernetes" />

## Prerequisites
Before you begin, make sure that you have:

1. At least 15 GB of free storage space.
2. 4 GB of RAM.
3. Installed Helm package manager on your local machine. See the official Helm documentation for your [operation system](https://helm.sh/docs/intro/install/#through-package-managers).
4. A running Kubernetes cluster. If not present, follow these steps to set up the cluster:

>     a. Verify if you have access to AWS EKS with:

Reviewer: ```suggestion
    a. Verify if you have access to AWS cli with:
```

2. 4 GB of RAM.

## Before you begin
1. Installed Helm package manager on your local machine. See the official Helm documentation for your [operation system](https://helm.sh/docs/intro/install/#through-package-managers).
2. A running Kubernetes cluster. If not present, follow these steps to set up the cluster:

    a. Verify if you have access to AWS CLI with:
      
    ```bash
      aws sts get-caller-identity 
    ```
    
    The above command provides information related to your account and ARN, indicating that you can connect and access your Amazon account using a terminal.

    b. Create KubeConfig with: 

    ```bash
    aws eks update-kubeconfig --region ap-south-1 --name CLUSTER_NAME  --profile PROFILE_NAME
    ```
    In the above command, add the profile name that has access to the EKS cluster to the `--profile` parameter.

    c. Test your Kubernetes configuration with:

    ```bash
    kubectl cluster-info
    ```
    The above command provides a summary of the current cluster configuration, including the Kubernetes master and other cluster information. 

    d. A default storage class running. If not created yet, create the storage class with:

    ```bash

>     kubectl apply -f - <<EOF

Author: @pratapaprasanna - Please update the storage file command based on our discussion.

Reviewer: **for Kubernetes Version <= 1.20
we can use this 
for Kubernetes Version >= 1.23**

```
helm repo add aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
helm repo update
helm upgrade --install aws-ebs-csi-driver \
    --namespace kube-system \
    aws-ebs-csi-driver/aws-ebs-csi-driver
```
**Verify the installation using** 
```
kubectl get pods -n kube-system -l app.kubernetes.io/name=aws-ebs-csi-driver
```

<VideoEmbed host="youtube" videoId="wZzYL1uZwds" title="How To Deploy Appsmith On Kubernetes" caption="How To Deploy Appsmith On Kubernetes" />

## Prerequisites

> 1. At least 15 GB of free storage space.

Reviewer 2: these can be just bullets

2. 4 GB of RAM.

## Before you begin

> 1. Installed Helm package manager on your local machine. See the official Helm documentation for your [operating system](https://helm.sh/docs/intro/install/#through-package-managers).

Reviewer 2: ```suggestion
1. Install Helm package manager on your local machine. See the official Helm documentation for your [operating system](https://helm.sh/docs/intro/install/#through-package-managers).
```

> 2. Installed and configured `kubectl` to interact with your Kubernetes cluster. Follow the below guides available on the official Kubernetes documentation for instructions on how to install `kubectl` on your specific operating system: 

Reviewer 2: ```suggestion
2. Install and configure `kubectl` to interact with your Kubernetes cluster. Follow the below guides available on the official Kubernetes documentation for instructions on how to install `kubectl` on your specific operating system: 
```

    * [MacOS using Homebrew](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-with-homebrew-on-macos)
    * [Linux using native package management](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management)
    * [Windows using Chocolatey, Scoop, or winget](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-nonstandard-package-tools).

> 3. A running Kubernetes cluster. If not present, follow these steps to set up the cluster on AWS EKS:

Reviewer 2: ```suggestion
3. Set up a Kubernetes cluster. Follow these steps to set up the cluster on AWS EKS:
```

  :::info

>    You are free to choose your preferred platform for hosting the Kubernetes cluster. Make sure you have configured Kubeconfig and created a default storage class before proceeding with the Appsmith installation. If you face any issues, reach out to the support team at [support@appsmith.com](mailto:support@appsmith.com).

Reviewer 2: we are recommending chat widget everwhere right?

Author: Yes, but here it makes sense for them to contact support using email, as they are having issues in their prerequisite setup. What do you think?

Reviewer 2: either way its going to be answered by Pranav and team only. The reason I'm thinking we should use the chat widget is to be able to easily measure how many people have issues and reach out from docs. Secondly its one less step for them to send an email

  :::

    a. Verify if you have access to AWS CLI with:
      
    ```bash
      aws sts get-caller-identity 
    ```
    
    The above command provides information related to your account and ARN, indicating that you can connect and access your Amazon account using a terminal.

    b. Create KubeConfig with: 

    ```bash
    aws eks update-kubeconfig --region ap-south-1 --name CLUSTER_NAME  --profile PROFILE_NAME
    ```
    In the above command, add the profile name that has access to the EKS cluster to the `--profile` parameter.

    c. Test your Kubernetes configuration with:

    ```bash
    kubectl cluster-info
    ```
    The above command provides a summary of the current cluster configuration, including the Kubernetes master and other cluster information. 

>     d. A default storage class running. If not created yet, create the storage class with:

Reviewer 2: ```suggestion
    4. Define storage class. If you haven't already, define the storage class with:
```

Author: It's not defining storage class. They have to create it.

>     ```bash

Reviewer 2: if we are providing them commands, we should always specify where they have run these.

Author: Just looked at it again, all the above steps are commands only, and they don't need to do it in a specific folder. Just on the terminal. I think it's intuitive that the users will open a terminal and we don't have to specify that. @pratapaprasanna - Please confirm.
